### PR TITLE
remove ubuntu 2304 images

### PR DESCRIPTION
**Pull request by a-crate**
/cc @zmarano @elicriffield

### CODE REVIEW

File: prow/prowjobs/GoogleCloudPlatform/gcp-guest/gcp-guest-config.yaml [hunk 1]
```diff
@@ -554,7 +554,7 @@ periodics:
...
.../images/family/ubuntu-pro-2004-lts,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-2204-lts,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-fips-1804-lts,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-fips-2004-lts,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-1804-lts-arm64,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-2004-lts-arm64,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-2204-lts-arm64,projects/ubuntu-os-cloud/global/images/family/ubuntu-2004-lts-arm64,projects/ubuntu-os-cloud/global/images/family/ubuntu-2204-lts-arm64,projects/ubuntu-os-cloud/global/images/family/ubuntu-minimal-2004-lts-arm64,projects/ubuntu-os-cloud/global/images/family/ubuntu-minimal-2204-lts-arm64,projects/ubuntu-os-cloud/global/images/family/ubuntu-minimal-2304-arm64,projects/windows-cloud/global/images/family/windows-2016,projects/windows-cloud/global/images/family/windows-2016-core,projects/windows-cloud/global/images/family/windows-2019,projects/windows-cloud/global/images/family/windows-2019-core,projects/windows-cloud/global/images/family/windows-2022,projects/windows-cloud/global/images/family/windows-2022-core"
```
Comment (gaughen) [inline prow/prowjobs/GoogleCloudPlatform/gcp-guest/gcp-guest-config.yaml:557]: cloud/global/images/family/ubuntu-minimal-2304-amd64 is still in the list. Thank you for giving me something to find.

Comment (gaughen) [reply prow/prowjobs/GoogleCloudPlatform/gcp-guest/gcp-guest-config.yaml:557]: and projects/ubuntu-os-cloud/global/images/family/ubuntu-minimal-2304-arm64

File: prow/prowjobs/GoogleCloudPlatform/gcp-guest/gcp-guest-config.yaml [hunk 2]
```diff
@@ -575,5 +575,5 @@ periodics:
...
...l/images/family/ubuntu-minimal-2004-lts,projects/ubuntu-os-cloud/global/images/family/ubuntu-minimal-2204-lts,projects/ubuntu-os-cloud/global/images/family/ubuntu-minimal-2304-amd64,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-1604-lts,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-1804-lts,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-2004-lts,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-2204-lts,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-fips-1804-lts,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-fips-2004-lts,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-1804-lts-arm64,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-2004-lts-arm64,projects/ubuntu-os-pro-cloud/global/images/family/ubuntu-pro-2204-lts-arm64,projects/ubuntu-os-cloud/global/images/family/ubuntu-2004-lts-arm64,projects/ubuntu-os-cloud/global/images/family/ubuntu-2204-lts-arm64,projects/ubuntu-os-cloud/global/images/family/ubuntu-minimal-2004-lts-arm64,projects/ubuntu-os-cloud/global/images/family/ubuntu-minimal-2204-lts-arm64,projects/ubuntu-os-cloud/global/images/family/ubuntu-minimal-2304-arm64"
```
Comment (gaughen) [inline prow/prowjobs/GoogleCloudPlatform/gcp-guest/gcp-guest-config.yaml:578]: the 2304 arm64 ones are still on the list here too.

**Review by gaughen — CHANGES_REQUESTED** [review]

**Comment by a-crate** [comment]
I tried to deduplicate some of the lists but it's not easily possible I think, yaml has no string interpolation and poor referencing primitives. The fact that prow does yaml -> json -> native struct marshalling probably isn't helping.

Anyway I removed the last references to ubuntu 2304, and organized the lists. Now they should have windows images at the front (because they're missing from one list, for easy comparison of the rest), ubuntu images at the back (because versions change most frequently), all images should be grouped by project, and within each project things should go in version order. Within a version it should be amd64 -> arm64 -> other variants.

jsonnet would make this easier..

**Review by gaughen — APPROVED** [review]
LGTM

**Pull request merged**